libs/wingc/src/jsify.rs

```diff
@@ -1060,22 +1060,36 @@ impl<'a> JSifier<'a> {
 			code.line("super(scope, id);");
 		}
 
-		if inflight_methods.len() + inflight_fields.len() > 0 {
-			let inflight_method_names = inflight_methods.iter().map(|(name, _)| name.name.clone()).collect_vec();
-			let inflight_field_names = inflight_fields.iter().map(|f| f.name.name.clone()).collect_vec();
-			let inflight_ops_string = inflight_method_names
-				.iter()
-				.chain(inflight_field_names.iter())
-				.map(|name| format!("\"{}\"", name))
-				.join(", ");
-			code.line(format!("this._addInflightOps({inflight_ops_string});"));
-		}
-
 		let init_statements = match &constructor.body {
 			FunctionBody::Statements(s) => s,
 			FunctionBody::External(_) => panic!("'init' cannot be 'extern'"),
 		};
 
+		if !no_parent {
+			// Check if the first statement is a super constructor call, if not we need to add one
+			let first_statement = if init_statements.statements.len() > 0 {
+				Some(&init_statements.statements[0])
+			} else {
+				None
+			};
+
+			match first_statement {
+				Some(s) => {
+					match s.kind {
+						StmtKind::SuperConstructor { arg_list: _ } => {} // Already has super do nothing
+						_ => code.line("super(scope, id);"),
+					}
+				}
+				None => {
+					// no statements so we implicitly call super since we know there is a parent
+					code.line("super(scope, id);")
+				}
+			}
```

Comment:
Here's attempt:

```rust
let super_call = if let Some(s) = init_statements.statements.first() {
	matches!(s.kind, StmtKind::SuperConstructor { .. })
} else {
	false
};

// no statements so we implicitly call super since we know there is a parent
if !super_call {
	code.line("super(scope, id);")
}
```

Reply:
Gorgeous!